source/auth/Credentials.cpp

```diff
@@ -265,7 +267,9 @@ namespace Aws
                 struct aws_credentials_provider_chain_default_options raw_config;
                 AWS_ZERO_STRUCT(raw_config);
 
-                raw_config.bootstrap = config.Bootstrap->GetUnderlyingHandle();
+                raw_config.bootstrap =
+                    config.Bootstrap ? config.Bootstrap->GetUnderlyingHandle()
+                                     : ApiHandle::GetOrCreateStaticDefaultClientBootstrap()->GetUnderlyingHandle();
```

Comment:
You need to add documentation to `CredentialsProviderChainDefaultConfig::Bootstrap` about how the default will be used if it's not set.